workflow-service/src/main/java/com/redhat/parodos/project/service/ProjectServiceImpl.java

```diff
@@ -54,6 +55,10 @@ public ProjectServiceImpl(ProjectRepository projectRepository, WorkFlowRepositor
 
 	@Override
 	public ProjectResponseDTO save(ProjectRequestDTO projectRequestDTO) {
+		if (projectRepository.findByName(projectRequestDTO.getName()).isPresent()) {
+			throw new EntityExistsException(
+					String.format("Project with name: %s already exists", projectRequestDTO.getName()));
+		}
 		// get user from security utils and set on project
 		Project project = projectRepository.save(Project.builder().name(projectRequestDTO.getName())
 				.description(projectRequestDTO.getDescription()).createDate(new Date()).modifyDate(new Date()).build());
```

Comment:
Moving forwards, what do you think of setting createDate and modifyDate via sysdate at the db level?

Reply:
That's a tricky one :-) 
It depends if the clock of the Postgres-db server is in sync with the workflow service.
If the importance is when the request is sent by the user - then I'd keep the current code.
If the importance is the actual record creation time - then use sysdate.

Reply:
Agreed with your points! Since we don't specific requirements on this, let's keep as-is :)
We may revisit later